services/management/grpc/actions_server.go

```diff
@@ -294,10 +295,64 @@ func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managemen
 	}, nil
 }
 
-// StartPTMySQLSummaryAction starts pt-mysql-summary action.
+func pointerToAgentType(agentType models.AgentType) *models.AgentType {
+	return &agentType
+}
+
+// StartPTMySQLSummaryAction starts pt-mysql-summary action and returns the pointer to the response message
 //nolint:lll
-func (s *actionsServer) StartPTMySQLSummaryAction(context.Context, *managementpb.StartPTMySQLSummaryActionRequest) (*managementpb.StartPTMySQLSummaryActionResponse, error) {
-	panic("not implemented yet")
+func (s *actionsServer) StartPTMySQLSummaryAction(ctx context.Context, req *managementpb.StartPTMySQLSummaryActionRequest) (*managementpb.StartPTMySQLSummaryActionResponse, error) {
+	// Need to get the service id's pointer to retrieve the list of agent pointers therefrom
+	// to get the particular agentID from the request.
+	service, err := models.FindServiceByID(s.db.Querier, req.ServiceId)
+	if err != nil {
+		return nil, err
+	}
+
+	pmmagents, err := models.FindPMMAgentsRunningOnNode(s.db.Querier, service.NodeID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "No pmm-agent running on this node")
+	}
+
+	pmmAgentID, err := models.FindPmmAgentIDToRunAction(req.PmmAgentId, pmmagents)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := models.CreateActionResult(s.db.Querier, pmmAgentID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Exporters to be filtered by service ID and agent type
+	agentFilter := models.AgentFilters{ServiceID: req.ServiceId, AgentType: pointerToAgentType(models.MySQLdExporterType)}
+
+	// Need to get the mysql exporters to get the username and password therefrom
+	mysqdExporters, err := models.FindAgents(s.db.Querier, agentFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	exportersCount := len(mysqdExporters)
+
+	// Must be only one result
+	if exportersCount != 1 {
```

Comment:
so we can split it to two ifs to not have nested ifs like
```
if exportersCount == 0 {
// return error here
}
if exportersCount > 2 {
// return error here
}
```

Reply:
We can. I didn't do it this way to be faster if there is no error. In your scenario you have to check for 0 and then for 2 and then continue. Two comparisons. 
In my scenario you just check that is 1. On comparison. And only check for 0 in the error branch where the efficient is less important than in case of no error.
If you disassemble it to instructions, you'll may find instructions to compare and conditional branch. Now depending on the machine, the branch can add stall cycles due to pipeline reload because the address changes. This depends on the CPU architecture. What I wanted here is to prevent unnecessary jumps.

Reply:
We don't call this method so often to care about that, but making the way I mentioned makes code much easier to read.